logfire/_internal/scrubbing.py

```
@@ -319,21 +464,25 @@ def scrub(self, path: JsonPath, value: Any) -> Any:
+                    return json.dumps(self.scrub(path, value, pattern))
         elif isinstance(value, Sequence):
-            return [self.scrub(path + (i,), x) for i, x in enumerate(cast('Sequence[Any]', value))]
+            return [self.scrub(path + (i,), x, pattern) for i, x in enumerate(cast('Sequence[Any]', value))]
         elif isinstance(value, Mapping):
             result: dict[str, Any] = {}
             for k, v in cast('Mapping[str, Any]', value).items():
-                if k in BaseScrubber.SAFE_KEYS:
+                if k in self._user_safe_keys:
+                    # A user safe key names arbitrary application data, so it is exempt from
+                    # everything except the patterns guarding Logfire's own token.
+                    result[k] = self.scrub(path + (k,), v, self._credential_pattern)
```

Ruff (0.16.1)

[warning] Consider `(*path, k)` instead of concatenation

Replace with `(*path, k)`

(RUF005)